bpf/lib/egress_gateway.h

```diff
@@ -43,7 +43,9 @@ int egress_gw_fib_lookup_and_redirect(struct __ctx_buff *ctx, __be32 egress_ip,
 	if (egress_ifindex && neigh_resolver_available())
 		return redirect_neigh(egress_ifindex, NULL, 0, 0);
 
-	ret = (__s8)fib_lookup_v4(ctx, &fib_params, egress_ip, daddr, 0);
+	ret = (__s8)fib_lookup_v4(ctx, &fib_params, egress_ip, daddr,
+				  neigh_resolver_available()
+				  ? BPF_FIB_LOOKUP_SKIP_NEIGH : 0);
```

Comment:
Just to reduce the churn in the patch, you could push this logic down into the `fib_lookup_v4()` and `fib_lookup_v6()` helpers.

Reply:
> Just to reduce the churn in the patch, you could push this logic down into the fib_lookup_v4() and fib_lookup_v6() helpers.

IIRC there was some reason I didn't do this, but I'll take another look and see if it's possible.

Reply:
I went ahead and pushed this into `fib_lookup_v(4|6)`. My main concern before was that setting the flag inside the function doesn't make sense if there are callers that don't immediately follow up with a call to `fib_do_redirect`. However, as far as I can tell, this seems to be the usage pattern all over. Feel free to take another look.

*Note*: Unit tests are passing, but looks like some CI issue is currently blocking image builds for smoke tests, conformance tests, etc. Filed https://github.com/cilium/cilium/issues/39600